### PR TITLE
Introduce a --registry-insecure flag

### PR DESCRIPTION
**Pull request by dprotaso**
- **fix env var casing otherwise flags with hyphens will continue to have hyphens**
- **Introduce an --registry-insecure flag**

### CODE REVIEW

**Comment by dprotaso** [comment]
/hold

Deploy is still broken.

Tthough I'm noticing 'insecure' in oci.Pusher code is being confused with `anonymous` user so I'm unsure what's best to do here

**Comment by dprotaso** [comment]
All good now - anon is still disabled by default (like it was before).

Insecure flag changes the registry scheme

File: cmd/root.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ Learn more about Knative at: https://knative.dev`, cfg.Name),
 	// a version prefixed by "FUNC_"
 	viper.AutomaticEnv()       // read in environment variables for FUNC_<flag>
 	viper.SetEnvPrefix("func") // ensure that all have the prefix
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
```
Comment (lkingland) [inline cmd/root.go:61]: Good stuff!  I am surprised this isn't the default behavior

File: cmd/root.go [hunk 2]
```diff
@@ -170,6 +171,7 @@ func bindEnv(flags ...string) bindFunc {
 		}
 		viper.AutomaticEnv()       // read in environment variables for FUNC_<flag>
 		viper.SetEnvPrefix("func") // ensure that all have the prefix
+		viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
```
Comment (lkingland) [inline cmd/root.go:174]: And this duplicate directive is another reason we have a flag refactor/cleanup in the works

**Review by lkingland — APPROVED** [review]

**Pull request merged**